drivers/input/input_chsc6x.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2024 Nicolas Goualard <nicolas.goualard@sfr.fr>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT seeed_chsc6x
+
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/input/input.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/logging/log.h>
+
+struct chsc6x_config {
+	struct i2c_dt_spec i2c;
+	const struct gpio_dt_spec int_gpio;
+};
+
+struct chsc6x_data {
+	const struct device *dev;
+	struct k_work work;
+	struct gpio_callback int_gpio_cb;
+};
+
+struct chsc6x_output {
+	uint8_t points;
+	uint16_t x;
+	uint16_t y;
+} __packed;
+
+LOG_MODULE_REGISTER(chsc6x, CONFIG_INPUT_LOG_LEVEL);
+
+static int chsc6x_process(const struct device *dev)
+{
+	uint16_t row, col;
+	bool is_pressed;
+
+	struct chsc6x_output output;
+	const struct chsc6x_config *cfg = dev->config;
+
+	int res = i2c_burst_read_dt(&cfg->i2c, 0, (uint8_t *)&output, sizeof(output));
+
+	if (res < 0) {
+		LOG_ERR("Could not read data %i", res);
+		return -ENODATA;
+	}
```

Comment:
```suggestion
	res = i2c_burst_read_dt(&cfg->i2c, 0, (uint8_t *)&output, sizeof(output));
	if (res < 0) {
		LOG_ERR("Could not read data %i", res);
		return -ENODATA;
	}
```

and declare `res` with the other automatic variables, also could you add a define for that `0`?

Reply:
oh and if you could `ret` instead of `res`, trying to keep the input drivers coherent, makes it easy to share code, thanks!